composables/use-subscription.ts

```diff
@@ -0,0 +1,96 @@
+import { PaywallModal } from '#components'
+
+export function useSubscription() {
+  const { t: $t } = useI18n()
+  const { user } = useUserSession()
+  const selectedPlan = ref('yearly')
+  const { loggedIn: hasLoggedIn } = useUserSession()
+  const accountStore = useAccountStore()
+  const isProcessingSubscription = ref(false)
+  const { handleError } = useErrorHandler()
+  const localeRoute = useLocaleRoute()
+  const toast = useToast()
+
+  const isLikerPlus = computed(() => {
+    if (!hasLoggedIn.value) return false
+    return user.value?.isLikerPlus
+  })
+
+  const overlay = useOverlay()
+  const paywallModal = overlay.create(PaywallModal, {
+    props: {
+      handleSubscribe: startSubscription,
+      isProcessingSubscription,
+    },
+  })
+
+  async function checkLikerPlusStatus() {
+    try {
+      if (!hasLoggedIn.value) {
+        await accountStore.login()
+        if (!hasLoggedIn.value) return
+      }
+      if (user.value?.isLikerPlus) {
+        navigateTo(localeRoute({ name: 'account' }))
+      }
+    }
+    catch (error) {
+      handleError(error)
+    }
+  }
+
+  async function startSubscription() {
+    await checkLikerPlusStatus()
+    useTrackEvent('subscription_button_click', { plan: selectedPlan.value })
```

Comment:
I think checkLikerPlusStatus should return a boolean so that can stop startSubscription, as i

```ts
const isSubscribed = await checkLikerPlusStatus()
if (isSubscribed) return
```